hw/top_earlgrey/data/placement_cw310.xdc

```diff
@@ -0,0 +1,60 @@
+############
+# Boot ROM #
+############
+# We need to enforce a defined placement of the Boot ROM to enable the Boot ROM splicing script.
+# Any change in ROM instances path should be updated in following two files
+# 1. hw/top_earlgrey/data/placement.xdc and
+# 2. hw/top_earlgrey/util/vivado_hook_opt_design_post.tcl
+set_property LOC RAMB36_X0Y10 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_0" && PRIMITIVE_TYPE =~ BMEM.*.* }]
+set_property LOC RAMB36_X0Y11 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_1" && PRIMITIVE_TYPE =~ BMEM.*.* }]
+set_property LOC RAMB36_X0Y12 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_2" && PRIMITIVE_TYPE =~ BMEM.*.* }]
+set_property LOC RAMB36_X0Y13 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_3" && PRIMITIVE_TYPE =~ BMEM.*.* }]
+set_property LOC RAMB36_X0Y14 [get_cells -hierarchical -filter { NAME =~ "*u_rom_ctrl*u_rom*rdata_o_reg_4" && PRIMITIVE_TYPE =~ BMEM.*.* }]
+
+###############
+# Guiding P&R #
+###############
+# Additional placement rules are for guiding P&R to:
+# - speed up implementation time, and
```

Comment:
Is this something you found out, or is it documented somewhere? It seems counterintuitive at first to me that an additional constraint makes placement easier (and thus faster).

Reply:
The idea is to let place and route start from a known good placement. This is actually recommended by Xilinx to speed up the routing, see https://www.xilinx.com/support/answers/66314.html
I should probably mention that.